contrib/pyln-testing/pyln/testing/fixtures.py

```diff
@@ -25,6 +25,8 @@
 @pytest.fixture(scope="session")
 def test_base_dir():
     d = os.getenv("TEST_DIR", "/tmp")
+    if len(d) > 12:
+        raise ValueError("TEST_DIR length is too long please create one with less than x characters")
```

Comment:
do you think that this is a hacky approach?

Maybe the solution is to fix the opts parser?

cc: @javiervargas that is working on a C solution

Reply:
ya its a bit hacky would you be able to point me to the opts parser? @javiervargas do you have a branch maybe I could look at?

Reply:
is this where I should be looking?

https://github.com/ElementsProject/lightning/blob/master/lightningd/options.c#L1420